### PR TITLE
Support free-form objects

### PR DESCRIPTION
**Pull request by ityuhui**
This PR merges the upstream PR https://github.com/OpenAPITools/openapi-generator/pull/12557 to support `free-form` objects.

And fixes https://github.com/kubernetes-client/c/issues/123

### CODE REVIEW

File: kubernetes/model/object.c [hunk 1]
```diff
@@ -4,28 +4,48 @@
...
+    if (!object) {
+        return NULL;
+    }
 
-    return item;
-fail:
-    cJSON_Delete(item);
-    return NULL;
+    if (!object->temporary) {
+        return cJSON_Parse("{}");
```
Comment (brendandburns) [inline kubernetes/model/object.c:31]: Is this actually correct? Wouldn't `cJSON_Parse("null")` be more appropriate?

**Comment by brendandburns** [comment]
One minor comment wrt what the default value should be, otherwise LGTM.

File: kubernetes/model/object.c (hunk 1 above)
Comment (ityuhui) [reply kubernetes/model/object.c]: "null" is more appropriate. I updated this.

**Comment by ityuhui** [comment]
Thanks @brendandburns 
I have updated the code.
And will merge back to upstream openapi-generator.

**Pull request merged**